astropy/coordinates/builtin_frames/utils.py

```diff
@@ -34,29 +36,42 @@
 _DEFAULT_PM = (0.035, 0.29)*u.arcsec
 
 
+def matrix_product(*args):
+    """Matrix multiply all arguments together.
+
+    Arguments should have dimension 2 or larger, with larger dimensional
+    objects being interpreted as stacks of matrices.
+    """
+    return reduce(matmul, args)
+
+
+def matrix_transpose(m):
+    """Transpose a matrix or stack of matrices by swapping the last two axes."""
+    return m.swapaxes(-2, -1)
+
+
 def cartrepr_from_matmul(pmat, coo, transpose=False):
     """
     Note that pmat should be an ndarray, *not* a matrix.
     """
```

Comment:
Should we include in the docstring here the way pmat is treated if it is ```N``` dimensional and ```N > 2```? I'm thinking of future ```coordinates``` developers trying to understand what shape their arrays should be...